### DISCUSSION TITLE
Does Mixcore provide Localization feature

### DISCUSSION THREAD

**Discussion by UmerTariqKhan** [mixcore/mix.core#393] [Q&A] [answered]
Does Mixcore provide a way to create localized versions of the content? For Example: If we Defined a language then we have an option to create the content of its language ??
![image_2021_05_30T21_45_34_870Z](https://user-images.githubusercontent.com/36357077/120120967-69eab780-c1ba-11eb-8d35-86b47a6099a2.png)

**Comment by Smilefounder** [comment, marked as answer]
Hi @UmerTariqKhan yes, Mixcore supports multi-languages by default. Just add a new language, switch to the target language then create content for that lang. :)

**Reply by UmerTariqKhan** [reply]
Thanks for the reply. Any documentation or example link will be more help full. 
You mean 
1- First I have to save with English by checking the checkbox from the language and save it.
2- Then again save by checking another language checkbox?
     Remaining in the same form?
 
 
![image_2021_05_31T09_18_30_009Z](https://user-images.githubusercontent.com/36357077/120171136-23797500-c21b-11eb-88af-384ad0e7b48a.png)

**Reply by Smilefounder** [reply]
@UmerTariqKhan yes, just tick check on the sidebar then switch the language on the top bar to update the content for specific lang.
![image](https://user-images.githubusercontent.com/3785721/120217557-b7226400-c262-11eb-8cf4-bcc9e7b03049.png)

Cheers!

**Comment by UmerTariqKhan** [comment]
Hi, I am facing an issue, I just add a new localization setting and add language as Arabic. When I just changed the language from the top. Textboxes were supposed to start from the right side as this language starts from RIGHT. Please see the attachment
![image_2021_05_31T20_44_45_434Z](https://user-images.githubusercontent.com/36357077/120240859-05dbf800-c27b-11eb-9d1c-8d50ca755b83.png)


> Does Mixcore provide a way to create localized versions of the content? For Example: If we Defined a language then we have an option to create the content of its language ??
> ![image_2021_05_30T21_45_34_870Z](https://user-images.githubusercontent.com/36357077/120120967-69eab780-c1ba-11eb-8d35-86b47a6099a2.png)

**Reply by Smilefounder** [reply]
Hi @UmerTariqKhan btw, if your language is mainly RTL at the moment you can do a quick trick as below:
1. Find the following file in your local source code https://github.com/mixcore/mix.core/blob/master/src/Mix.Cms.Web/Views/Shared/_LayoutPortal_Angular.cshtml
2. Set dir="rtl" on the <html> element. (ref: https://getbootstrap.com/docs/5.0/getting-started/rtl/)

That all : )

![image](https://user-images.githubusercontent.com/3785721/120263045-2466e100-c2c5-11eb-951b-f60da0ef07b8.png)

**Comment by Smilefounder** [comment]
Hi @UmerTariqKhan from the Admin Portal, we do not support RTL UI yet. However, you still able to input the content and show it to the web theme with RTL display. Sorry for the inconvenience. We will take note of this and enhance it asap.

cc @mixcore/core-team

**Reply by UmerTariqKhan** [reply]
Ok. Thanks for your support. 
I am again facing an issue.
1- I changed the language of the portal to Arabic. Then I wrote the content in Arabic.
2- From Post List, When I click on the eye button it shows me content in Arabic which is right.
3- On the eye Dialog when I click on the new window, it shows the content in English. Please have a look at the attachments.
Thanks
![image_2021_06_01T11_28_45_267Z](https://user-images.githubusercontent.com/36357077/120316152-8a1f9100-c2f6-11eb-9caf-985cb78187bf.png)
![Inkedimage_2021_06_01T11_17_14_937Z_LI](https://user-images.githubusercontent.com/36357077/120316165-8d1a8180-c2f6-11eb-90ec-d2113d4c5342.jpg)
![Inkedimage_2021_06_01T11_17_57_853Z_LI](https://user-images.githubusercontent.com/36357077/120316291-b20ef480-c2f6-11eb-9510-5a478a0f91a1.jpg)

**Reply by Smilefounder** [reply]
ah so sorry for the confusion, our current themes don't support switch languages at the moment. I thought you are a developer so I didn't mention that.

In case you would like to modify the theme to show the other languages, let me grab some guidelines for you.

**Reply by UmerTariqKhan** [reply]
I applied the fix you suggest above, on the admin porta, but that doesn't look nice. As this thing not supported. It's okay, But the Theme should have the language switching functionality e.g: RTL LTL. If yes please provide steps or a link.

On theme:
1- I Need a dropdown for languages.
2- Apply them accordingly e.g: RTL, LTL.

**Reply by UmerTariqKhan** [reply]
@Smilefounder Huy Nguyen Thank you soo much for your support. I Liked your amazing product and priceless effort on it.

1- I am struggling at this point and needed help, I want a dropdown for languages like the below attachment. 
    Should I have to write custom methods in MixCmsHelper? To fetch languages and trigger their change, RTL and LTL?

Any help will be appreciated.
Thanks

![image](https://user-images.githubusercontent.com/36357077/120396254-ae5b8c00-c34f-11eb-9c42-60ad7ddfea22.png)
![image](https://user-images.githubusercontent.com/36357077/120395814-1067c180-c34f-11eb-8697-1f817c538d6e.png)
![image](https://user-images.githubusercontent.com/36357077/120396039-6177b580-c34f-11eb-82c2-97977503c07a.png)

**Reply by Smilefounder** [reply]
Hi @UmerTariqKhan I and my team @mixcore/core-team will get back to you asap. Sorry for the late response. We are stuck on some of the other offices works at the moment.

**Reply by Smilefounder** [reply]
btw, could you try to deploy your code to any publish domain? we might access it directly and support it in case we can.

**Reply by nhathoang989** [reply]
Hi @UmerTariqKhan,
In case you want to get supported languages, there are several way to get it:

- From Razor View:
 `var supportedCultures = await Mix.Cms.Lib.ViewModels.MixCultures.ReadViewModel.Repository.GetModelListAsync();`
- From Api:
 Get: https://yourdomain.com/api/v1/rest/shared/get-shared-settings
 
![image](https://user-images.githubusercontent.com/5327699/120523342-fafea000-c3ff-11eb-9275-7e384b5b5149.png)

**Reply by nhathoang989** [reply]
Hi @UmerTariqKhan 

Please pull the latest source code from the master, there're issues when loading content by language.

#396
#397 
#398 

Then check the following Sample Widget to loop your supported languages.


![image](https://user-images.githubusercontent.com/5327699/120529218-73685f80-c406-11eb-8de7-c34e40dae84b.png)


 Or you can take a look at [Dev Mixcore](https://dev.mixcore.org/en-us/post/4/top-open-source-asp-net-core-content-management-system-cms) for example

**Comment by UmerTariqKhan** [comment]
Thanks for the support. I will deploy tomorrow and let you know.

**Comment by UmerTariqKhan** [comment]
Thank you soo much for your support. Great Its working. Wish if LTR, RTL
also be fixed.

Thanks

**Reply by Smilefounder** [reply]
Thanks @UmerTariqKhan for letting us know. And thanks a lot for using Mixcore :)

**Comment by UmerTariqKhan** [comment]
To implement LTR and RTL in the theme,  should I set another ViewBag like
(lang="@ViewBag.culture") ??

**Reply by Smilefounder** [reply]
yes, also using this CSS inside html tag
```
direction: rtl;
```
https://www.w3schools.com/cssref/pr_text_direction.asp

**Comment by UmerTariqKhan** [comment]
Hi Huy Nguyen, I am exploring Mix core and willing to set up a CMS app. I
am stuck at a point just created a module and unable to fetch its data
using the model. Please help

Thanks

**Reply by Smilefounder** [reply]
@UmerTariqKhan kindly jump over to this thread https://github.com/mixcore/mix.core/discussions/402
